src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -448,8 +448,7 @@ ciKlass* ciEnv::get_klass_by_name_impl(ciKlass* accessing_klass,
       kls = SystemDictionary::find_constrained_instance_or_array_klass(sym, loader,
                                                                        CHECK_AND_CLEAR_(fail_type));
     } else {
```

Comment:
I think SystemDictionary::find_constrained_instance_or_array_klass can also be changed to accept a `Thread*` instead `TRAPS`, since now it can no longer throw, and the thread is used only for Mutexes.

Reply:
Hi Ioi, I'd like to do this in a separate RFE.  Is that okay?